src/message.ts

```diff
@@ -42,7 +42,9 @@ export interface MsgRawObj {
   MMActualSender:   string, // getUserContact(message.MMActualSender,message.MMPeerUserName).isContact()
   MMPeerUserName:   string, // message.MsgType == CONF.MSGTYPE_TEXT && message.MMPeerUserName == 'newsapp'
   ToUserName:       string,
+  FromUserName:     string,
```

Comment:
When MsgType in [1,3,43,47,49],FromUserName and Content exist..
Other unconfirmed.